basilica/basilica/__init__.py

```diff
@@ -81,16 +86,60 @@ def raw_embed(self, url, data, opts, timeout):
     # TODO: parallelize
     def embed(self, url, data, batch_size, opts, timeout):
         batch = []
+        outstanding = 0
+        api_thread = threading.Thread(target=self.raw_embed_wrapper, args=(url, opts, timeout))
+        api_thread.setDaemon(True)
+        api_thread.start()
         for i in data:
             batch.append(i)
             if len(batch) >= batch_size:
-                for e in self.raw_embed(url, batch, opts=opts, timeout=timeout):
-                    yield e
+                self.write_queue.put_nowait(batch)
```

Comment:
I don't think `put_nowait` is what we want here.  We're constructing an infinite-size queue, so this is identical to just calling `put`.